src/main/kotlin/com/hibob/academy/dao/OwnerDao.kt

```diff
@@ -0,0 +1,58 @@
+package com.hibob.academy.dao
+
+import org.jooq.DSLContext
+import org.jooq.RecordMapper
+import org.jooq.Record
+
+class OwnerDao(private val sql: DSLContext) {
+    private val owner = OwnerTable.instance
+    private val pet = PetTable.instance
+
+    private val ownerDataMapper = RecordMapper<Record,OwnerData>{ record ->
+            OwnerData(
+                record[owner.id],
+                record[owner.ownerName],
+                record[owner.employeeId],
+                record[owner.companyId]
+            )
+    }
+
+    //Create a function that retrieve all the owners from the owner table and return their name, employee id and company id
+    fun getAllOwners(companyId: Long): List<OwnerData>{
+        return sql.select(owner.id, owner.ownerName,owner.employeeId,owner.companyId)
+            .from(owner)
+            .where(owner.companyId.eq(companyId))
+            .fetch(ownerDataMapper)
+    }
+
+//    Add a new API that get pet id and  return the owner’s information.
+    fun getOwnerByPetId(petId: Long, companyId: Long): OwnerData?{
+        return sql.select(owner.id, owner.ownerName,owner.employeeId,owner.companyId)
+            .from(pet)
+            .rightJoin(owner).on(pet.ownerId.eq(owner.id))
```

Comment:
Do rightJoin really the right join operation here?
What are the differences between the joins?

Reply:
Oops my bad!
I should use a left join because it ensures that the query returns the pet even if it doesn’t have an associated owner, which matches your goal of always retrieving the pet’s information.